### PR TITLE
Improved gauge, timing and value macro doc

### PR DESCRIPTION
**Pull request by jean-airoldie**
Further doc improvements.

### CODE REVIEW

File: metrics/src/macros.rs [hunk 1]
```diff
@@ -122,8 +130,7 @@ macro_rules! gauge {
 /// Labels can also be passed along:
 ///
 /// ```rust
-/// # #[macro_use]
-/// # extern crate metrics;
+/// # use metrics::timing;
```
Comment (tobz) [inline metrics/src/macros.rs]: I think we should remove the `#` so that this line is shown?

File: metrics/src/macros.rs [hunk 2]
```diff
@@ -86,13 +90,17 @@ macro_rules! gauge {
 
 /// Records a timing.
 ///
+/// This will register an histogram with the given name, if it does not already
+/// exist, then add data point with the given timing. This timing must implement
+/// `AsNanoseconds`. Optionally, a set of labels, of the form `key => value`,
```
Comment (tobz) [inline metrics/src/macros.rs]: You have to wrap the type in brackets -- ``[`AsNanoseconds`]`` -- for it to become a hyperlink.

File: metrics/src/macros.rs [hunk 3]
```diff
@@ -172,6 +181,10 @@ macro_rules! timing {
 
 /// Records a value.
 ///
+/// This will register an histogram with the given name, if it does not already
+/// exist, then add data point with the given value. Optionally, a set of labels,
+/// of the form `key => value`, can be passed to further describe the counter.
```
Comment (tobz) [inline metrics/src/macros.rs]: `counter` -> `value`

File: metrics/src/macros.rs [hunk 4]
```diff
@@ -86,13 +90,17 @@ macro_rules! gauge {
 
 /// Records a timing.
 ///
+/// This will register an histogram with the given name, if it does not already
+/// exist, then add data point with the given timing. This timing must implement
+/// `AsNanoseconds`. Optionally, a set of labels, of the form `key => value`,
+/// can be passed to further describe the counter.
```
Comment (tobz) [inline metrics/src/macros.rs]: `counter` -> `timing`

File: metrics/src/macros.rs [hunk 5]
```diff
@@ -43,27 +43,31 @@ macro_rules! counter {
...
 }
 
-/// Records a gauge.
+/// Update a gauge by a value.
 ///
-/// Functionally equivalent to calling [`Recorder::record_gauge`].
+/// This will register a gauge with the given name, if it does not already
+/// exist, then update it by the given value. A positive value will increment it,
+/// while a negative value will decrement it. Optionally, a set of labels,
+/// of the form `key => value`, can be passed to further describe the counter.
```
Comment (tobz) [inline metrics/src/macros.rs]: `counter` -> `gauge`

File: metrics/src/macros.rs (hunk 2 above)
Comment (jean-airoldie) [reply metrics/src/macros.rs]: a linting tool for dead links would be a godsend.

**Comment by jean-airoldie** [comment]
After testing, I found out that I made a mistake in the doc. The gauge! macro indeed takes an absolute value, so ill update that.

**Pull request merged**